Ot 41 amazon imagenes

devmontini opened the pull request.

rich27a committed:
AWS KEYS added to .example.env

rich27a committed:
aws-sdk added to config, aws.config updated, s3Bucket created

rich27a committed:
prettier enabled

rich27a committed:
package-lock added to gitignore

rich27a committed:
package-lock added to gitignore

devmontini committed:
Merge branch 'FIX-Categories_routes' into OT-41_Amazon_Imagenes

devmontini committed:
se mergeo el aws S3 de la rama de aws

devmontini committed:
AWS S3 Uplad Function:
Se creo la funcion de test y de upload file
Se modificaron algunas configuraciones de aws
Se agrego un file de prueba

soyarielruiz commented on .env.example:
```diff
@@ -1,6 +1,10 @@
+AWS_ACCESS_KEY_ID=
+AWS_SECRET_ACCESS_KEY=
+AWS_BUCKET_REGION=us-east-1
+AWS_BUCKET_NAME=cohorte-septiembre-5efe33c6
```
Estos datos NO deben estar en el repo.

soyarielruiz commented on .gitignore:
```diff
@@ -1,3 +1,4 @@
 node_modules
 .env
 docker-compose.yaml
+package-lock.json
```
Mala práctica no trackear _package-lock.json_

soyarielruiz commented on files/prueba.txt:
```diff
@@ -0,0 +1 @@
+Aguante el corte g291-node!
```
Esta prueba no debe subirse, debes tenerla vos para la demo. Sumado a que debe ser un archivo de imagen.

soyarielruiz commented on prettier.config.js:
```diff
@@ -9,4 +9,3 @@ const options = {
 };
 
 module.exports = options;
-
```
nit: Este borrado de última línea no se recomienda.

soyarielruiz commented on routes/index.js:
```diff
@@ -1,7 +1,7 @@
 var router = express.Router();
 const authRouter = require('./authentication');
-const category = require('./category');
+const categories = require('./categories');
```
nit: Estás retocando nombres innecesariamente, ya que es un punto de vista.

soyarielruiz commented on services/awsServices.js:
```diff
@@ -0,0 +1,40 @@
+require('dotenv').config();
+const s3 = require('../config/aws-sdk'); // Helper function that creates an Amazon S3 service client module.
```
nit: Comentarios que ensucian el código.

soyarielruiz commented on services/awsServices.js:
```diff
@@ -0,0 +1,40 @@
+
+      return s3.upload(uploadParams).promise();
+    } catch (error) {
+      res.status(400).json(error);
```
¿Porqué esta función debe devolver un status y testBucket un error?

soyarielruiz replied:
Esto debería lanzar una excepción, no un http status 400.

devmontini committed:
merged and resolve conflit

devmontini committed:
Merge branch 'development' into OT-41_Amazon_Imagenes

devmontini committed:
Se sacarpon las keys de AWS Service

devmontini committed:
Se corrigio awsS3:
Se mejoro la limpiaza del codigo
Se saco file/prueba.txt
Se sacaron algunos comentarios

soyarielruiz commented on config/aws-sdk.js:
```diff
@@ -0,0 +1,14 @@
+require('dotenv').config();
+const S3 = require('aws-sdk/clients/s3');
+
+const accessKey = process.env.AWS_ACCESS_KEY_ID;
```
Este file ya existe.

devmontini replied:
Este file no estaba mergeado en development. Decidi pasarme el file que hizo ricardo(con su permiso) a mi rama y luego trabajar el servicio de subida de archivos desde mi rama.

soyarielruiz replied:
No podes colocar código de otro PR que no se haya mergeado, es mala práctica y hace que no pueda mergearlo.

devmontini committed:
Merge branch 'development' into OT-41_Amazon_Imagenes

devmontini committed:
El servicio de aws s3 ahora lanza un error y no un status
